pnet_packet/src/dhcp.rs.in

```diff
@@ -0,0 +1,130 @@
+
+use PrimitiveValues;
+
+use pnet_macros_support::types::*;
+
+use std::net::Ipv4Addr;
+
+use pnet_base::MacAddr;
+
+/// Represents an Dhcp operation.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct DhcpOperation(pub u8);
+
+impl DhcpOperation {
+    /// Create a new `ArpOperation`.
+    pub fn new(value: u8) -> Self {
+        DhcpOperation(value)
+    }
+}
+
+impl PrimitiveValues for DhcpOperation {
+    type T = (u8,);
+    fn to_primitive_values(&self) -> (u8,) {
+        (self.0,)
+    }
+}
+
+/// The Dhcp protocol operations.
+#[allow(non_snake_case)]
+#[allow(non_upper_case_globals)]
+pub mod DhcpOperations {
+    use super::DhcpOperation;
+
+    /// DHCP request
+    pub const Request: DhcpOperation = DhcpOperation(1);
+
+    /// Dhcp reply
+    pub const Reply: DhcpOperation = DhcpOperation(2);
+}
+
+/// Represents the Dhcp hardware types.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct DhcpHardwareType(pub u8);
```

Comment:
Shouldn't this reuse the values from arp hardware?  

For reference, pls see: https://tools.ietf.org/html/rfc5494

Reply:
@pvinci: Feel free to be less humble  and/or less modest.

If your message is
> In section _2.  IANA Considerations_ of https://tools.ietf.org/html/rfc5494 is `ar$hrd (16 bits) Hardware address space`, that will not fit the `u8` of `DhcpHardwareType`.

just say so.

Reply:
Hardware type has 16 bit for DHCP according to [RFC-2131](https://www.ietf.org/rfc/rfc2131.txt). There are 2 fields in author struct: 8-bit for hwtype and 8-bit for hwlen.
I understand your point: ARP hwtype has 16-bit AND 8-bit for len, but from the point of dhcp protocol current HW fields are correct too, isn't?

Reply:
@stappersg, I don't understand your comment.  My comment was intended to be open ended to start a discussion.  My suggestion is that we just use the lower 8 bits of the existing list and avoid the hygiene related effort of maintaining multiple types.  Practically speaking, the list is small and changes infrequently.  

@biogen,
To answer your question, htype is 1 octet in RFC 2131.  Top of page 9:
```
htype         1  Hardware address type, see ARP section in "Assigned
                    Numbers" RFC; e.g., '1' = 10mb ethernet.
```
hwlen is the number of octets in the hw address, ie 6 for a mac address.

For further clarification on the values of hwlen, see rfc5494, sec. 2. Values for dhcp will be assigned to fit in a u8.
```
The following rules apply to the fields of ARP:

   ar$hrd (16 bits) Hardware address space

      Requests for ar$hrd values below 256 or for a batch of more than
      one new value are made through Expert Review [RFC5226].

      Note that certain protocols, such as BOOTP and DHCPv4, employ
      these values within an 8-bit field.  The expert should determine
      that a need to allocate the new values exists and that the
      existing values are insufficient to represent the new hardware
      address types.  
```